etcdserver/server.go

```diff
@@ -147,9 +147,14 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	if err != nil {
 		return nil, err
 	}
-	if walVersion == wal.WALUnknown {
+	switch walVersion {
+	case wal.WALUnknown:
 		return nil, fmt.Errorf("unknown wal version in data dir %s", cfg.DataDir)
+	case wal.WALv2_0Proxy:
+		// skip the discovery path and return the previous full cluster error again.
+		return nil, discovery.ErrFullCluster
```

Comment:
i do not think it is wired. we simply by pass the checking again and return the correct error it should return.